### PR TITLE
Files course copy small refactors

### PR DESCRIPTION
**Pull request by marcospri**
This tackles a few things mentioned in reviews before we introduce the new helper for groups:

- Makes the existing helper a service. This avoids having to pass a service around.
- Moves `extra` handling to he model
- Uses sane parameter names in the empty implementation of the plugin.

# Testing 

No behavior changes here, as a sanity check:

- make devdata
- tox -qe dockercompose -- exec postgres psql -U postgres -c "truncate grouping cascade;"

- Launch the scratch assignments and get a list of D2L and blackboard files in the original courses:

https://aunltd-test.blackboard.com/webapps/blackboard/execute/blti/launchPlacement?blti_placement_id=_20_1&content_id=_183_1&course_id=_19_1

https://aunltd.brightspacedemo.com/d2l/le/content/6782/viewContent/2431/View

Try these original courses:

https://aunltd.brightspacedemo.com/d2l/le/content/6782/viewContent/2374/View

https://aunltd-test.blackboard.com/webapps/blackboard/execute/blti/launchPlacement?blti_placement_id=_20_1&content_id=_39_1&course_id=_19_1

And the copied versions:

https://aunltd.brightspacedemo.com/d2l/le/content/6855/viewContent/2421/View

https://aunltd-test.blackboard.com/webapps/blackboard/execute/blti/launchPlacement?blti_placement_id=_20_1&content_id=_1371_1&course_id=_139_1

### CODE REVIEW

File: lms/models/grouping.py [hunk 1]
```diff
@@ -160,6 +160,20 @@ class D2LGroup(Grouping):
 class Course(Grouping):
     __mapper_args__ = {"polymorphic_identity": Grouping.Type.COURSE}
 
+    def get_mapped_file_id(self, file_id):
```
Comment (marcospri) [inline lms/models/grouping.py:163]: Moved from the service. `extra` is part of the model and I'd prefer to just take a look here to see what possibly could `extra` contain.

File: lms/services/__init__.py [hunk 2]
```diff
@@ -115,3 +115,14 @@ def includeme(config):
...
+
+    # Plugins are not the same as top level services but we want to register them as pyramid services too
+    # Importing them here to:
+    # - Don't pollute the lms.services namespace
+    # - Ease some circular-dependency problems
+    # pylint:disable=import-outside-toplevel
+    from lms.product.plugin.course_copy import CourseCopyFilesHelper
+
+    config.register_service_factory(
+        CourseCopyFilesHelper.factory, iface=CourseCopyFilesHelper
```
Comment (marcospri) [inline lms/services/__init__.py:127]: CourseCopyFilesHelper now a service itself.

File: lms/services/__init__.py [hunk 3]
```diff
@@ -115,3 +115,14 @@ def includeme(config):
     config.register_service_factory(
         "lms.services.d2l_api.d2l_api_client_factory", iface=D2LAPIClient
     )
+
+    # Plugins are not the same as top level services but we want to register them as pyramid services too
+    # Importing them here to:
+    # - Don't pollute the lms.services namespace
+    # - Ease some circular-dependency problems
+    # pylint:disable=import-outside-toplevel
```
Comment (marcospri) [inline lms/services/__init__.py:123]: This is not very nice but I rather not embark in a big lms/services refactor.

File: lms/product/plugin/course_copy.py [hunk 4]
```diff
@@ -87,11 +76,5 @@ class CourseCopyPlugin:  # pragma: nocover
...
         raise NotImplementedError()
 
-    def find_matching_file_in_course(self, *args, **kwargs):
-        raise NotImplementedError()
-
-    def get_mapped_file_id(self, *args, **kwargs):
-        raise NotImplementedError()
-
-    def set_mapped_file_id(self, course, old_file_id, new_file_id):
+    def find_matching_file_in_course(self, original_file_id, new_course_id):
```
Comment (marcospri) [inline lms/product/plugin/course_copy.py:79]: Changed the parameter names here

**Review by seanh — APPROVED** [review]

**Pull request merged**